src/operator/numpy/np_nonzero_op-inl.h

```diff
@@ -43,6 +43,25 @@ namespace mxnet {
 namespace op {
 
 struct NonzeroForwardKernel {
+  // this is for cpu
+  template<int ndim>
+  MSHADOW_XINLINE static void Map(index_t i,
+                                  int64_t* out,
```

Comment:
int64_t -> index_t

Reply:
This int64_t is actually the output data type. According to the original author this "must be int 64" https://github.com/apache/incubator-mxnet/blob/a2b77fb311204488bd72f99b681fc0ff334bca8d/src/operator/numpy/np_nonzero_op.cc#L33. 
I think the reasoning is because the tensor is flattened, even with int64tensorsize=off we can have > int32_max many elements e.g.(2**20, 2**20)

Reply:
For me, I feel that it's better to use template because most of the codes are duplicated.